### PR TITLE
NMS-13926: Fix Netflow option template handling

### PR DESCRIPTION
**Pull request by swachter**
Issue: https://issues.opennms.org/browse/NMS-13926

This PR

* fixes setting `templateTimeout` on Netflow parsers
* preserves Netflow option values in case of option template updates

### CODE REVIEW

File: features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/session/UdpSessionManager.java [hunk 1]
```diff
@@ -48,13 +49,13 @@
...
 import org.opennms.netmgt.telemetry.protocols.netflow.parser.state.ParserState;
+import org.opennms.netmgt.telemetry.protocols.netflow.parser.state.TemplateState;
 
 import com.google.common.collect.Iterables;
 import com.google.common.collect.Maps;
-import org.opennms.netmgt.telemetry.protocols.netflow.parser.state.TemplateState;
 
 public class UdpSessionManager {
-    final Map<TemplateKey, TimeWrapper<TemplateOptions>> templates = Maps.newConcurrentMap();
+    final ConcurrentMap<TemplateKey, TimeWrapper<TemplateOptions>> templates = Maps.newConcurrentMap();
```
Comment (fooker) [inline features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/session/UdpSessionManager.java:58]: Has this change a particular reason or is it just to emphasize that concurrency is impotent?

**Review by fooker — APPROVED** [review]
only one optional change.

File: opennms-util/src/main/java/org/opennms/core/utils/properties/DurationPropertyEditor.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2011-2022 The OpenNMS Group, Inc.
```
Comment (christianpape) [inline opennms-util/src/main/java/org/opennms/core/utils/properties/DurationPropertyEditor.java:4]:
```suggestion
 * Copyright (C) 2022 The OpenNMS Group, Inc.
```

**Review by christianpape — APPROVED** [review]
Nice.

File: features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/session/UdpSessionManager.java (hunk 1 above)
Comment (swachter) [reply features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/session/UdpSessionManager.java:58]: The method `ConcurrentMap.compute` that is used when updating the template is defined on `ConcurrentMap`.

File: features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/session/UdpSessionManager.java [hunk 3]
```diff
@@ -198,6 +199,11 @@ public TemplateOptions(final Template template) {
             this.template = Objects.requireNonNull(template);
             this.options = Maps.newConcurrentMap();
         }
+
+        public TemplateOptions(final Template template, Map<Set<Value<?>>, TimeWrapper<List<Value<?>>>> options) {
+            this.template = Objects.requireNonNull(template);
+            this.options = options;
```
[the remark this answers is not in the document]
Comment (swachter) [reply features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/session/UdpSessionManager.java:205]: Ok, added the check.

**Review by christianpape — APPROVED** [review]
Looks good.

**Pull request merged**